modules/fluid_properties/include/userobjects/FlibeFluidProperties.h

```diff
@@ -0,0 +1,286 @@
+   * @return temperature (K)
+   */
+  virtual Real T_from_v_e(Real v, Real e) const override;
+
+  /**
+   * Temperature and its derivatives from specific volume and specific internal energy
+   *
+   * @param[in] v        specific volume (m$^3$/kg)
+   * @param[in] e        specific internal energy (J/kg)
+   * @param[out] T       temperature (K)
+   * @param[out] dT_dv   derivative of temperature w.r.t. specific volume
+   * @param[out] dT_de   derivative of temperature w.r.t. specific internal energy
+   */
+  virtual void T_from_v_e(Real v, Real e, Real & T, Real & dT_dv, Real & dT_de) const override;
+
+  using SinglePhaseFluidProperties::cp_from_v_e;
+
+  /**
+   * Isobaric specific heat from specific volume and specific internal energy
+   *
+   * @param[in] v   specific volume (m$^3$/kg)
+   * @param[in] e   specific internal energy (J/kg)
+   * @return isobaric specific heat (J/kg.K)
+   */
+  virtual Real cp_from_v_e(Real v, Real e) const override;
+
+  using SinglePhaseFluidProperties::cv_from_v_e;
+
+  /**
+   * Isochoric specific heat from specific volume and specific internal energy
+   *
+   * @param[in] v   specific volume (m$^3$/kg)
+   * @param[in] e   specific internal energy (J/kg)
+   * @return isochoric specific heat (J/kg.K)
+   */
+  virtual Real cv_from_v_e(Real v, Real e) const override;
+
+  using SinglePhaseFluidProperties::mu_from_v_e;
+
+  /**
+   * Dynamic viscosity from specific volume and specific internal energy
+   *
+   * @param[in] v   specific volume (m$^3$/kg)
+   * @param[in] e   specific internal energy (J/kg)
+   * @return dynamic viscosity (Pa.s)
+   */
+  virtual Real mu_from_v_e(Real v, Real e) const override;
+
+  using SinglePhaseFluidProperties::k_from_v_e;
+
+  /**
+   * Thermal conductivity from specific volume and specific internal energy
+   *
+   * @param[in] v   specific volume (m$^3$/kg)
+   * @param[in] e   specific internal energy (J/kg)
+   * @return thermal conductivity (W/m.K)
+   */
+  virtual Real k_from_v_e(Real v, Real e) const override;
+
+  /**
+   * Density from pressure and temperature
+   *
+   * @param[in] p   pressure (Pa)
+   * @param[in] T   temperature (K)
+   * @return density (kg/m$^3$)
+   */
+  virtual Real rho_from_p_T(Real p, Real T) const override;
+
+  /**
+   * Density and its derivatives from pressure and temperature
+   *
+   * @param[in] p          pressure (Pa)
+   * @param[in] T          temperature (K)
+   * @param[out] rho       density (kg/m$^3$)
+   * @param[out] drho_dp   derivative of density w.r.t. pressure
+   * @param[out] drho_dT   derivative of density w.r.t. temperature
+   */
+  virtual void
+  rho_from_p_T(Real p, Real T, Real & rho, Real & drho_dp, Real & drho_dT) const override;
+
+  /**
+   * Specific volume from pressure and temperature
+   *
+   * @param[in] p   pressure (Pa)
+   * @param[in] T   temperature (K)
+   * @return specific volume (m$^3$/kg)
+   */
+  virtual Real v_from_p_T(Real p, Real T) const override;
```

Comment:
Can you wrap the two methods (`v_from_p_T`) in:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Woverloaded-virtual"
...
#undef propfunc
#pragma GCC diagnostic pop
```

That should fix the gcc problem.

The other solution would be to provide
```
virtual DualReal SinglePhaseFluidProperties::v_from_p_T(const DualReal&, const DualReal&) const
```

Reply:
That seems to have fixed the problem with the 3-parameter overload, but not the 5-parameter overload...

Reply:
I think you fixed all problems in the Flibe class. The error message is now:
```
In file included from /opt/civet/build_0/moose/modules/fluid_properties/build/header_symlinks/MultiComponentFluidProperties.h:13,
                 from /opt/civet/build_0/moose/modules/fluid_properties/build/header_symlinks/BrineFluidProperties.h:12,
                 from /opt/civet/build_0/moose/modules/fluid_properties/src/userobjects/BrineFluidProperties.C:10,
                 from /opt/civet/build_0/moose/modules/fluid_properties/build/unity_src/userobjects_Unity.C:2:
/opt/civet/build_0/moose/modules/fluid_properties/build/header_symlinks/SinglePhaseFluidProperties.h:224:20: error: 'virtual DualReal SinglePhaseFluidProperties::v_from_p_T(const DualReal&, const DualReal&) const' was hidden [-Werror=overloaded-virtual]
   virtual DualReal v_from_p_T(const DualReal & p, const DualReal & T) const;
```

That does not have flibe, but has the multi component fluid. I am not sure why this is happening...

Reply:
Seems ok to me!